src/lib.rs

```diff
@@ -177,7 +184,7 @@ cfg_if! {
         }
     } else {
         unsafe fn guess_os_stack_limit() -> usize {
-            panic!("cannot guess the stack limit on this platform");
+            0
```

Comment:
This may be breaching a bit too much into the realm of pedantry, but this causes `remaining_stack()` to return some suspicious results. Perhaps the existing `remaining_stack()` function could be updated though to return an `Option`?

Reply:
This actually makes a lot of things easier, since we now have a fast-path to recurse instead of going through the fallback functions.